packages/web/lib/page.js

```diff
@@ -4,18 +4,69 @@ const {
   resolveInternal,
   resolveFn,
   resolveFnScope,
+  resolveInternalScope,
+  validateScope,
 } = require("@browselang/core/lib/scope");
+const { evalRule, evalRuleSet, getNewScope } = require("@browselang/core");
 const { help } = require("@browselang/core/lib/utils");
+const fs = require("fs-extra");
+const { keys } = require("./constants");
+
+const dataStorageFn = (jsProcessing, type, optional = false) => (
+  scope
+) => async (key, selector) => {
+  validateScope((scope) => scope.internal.isPage, scope, true);
+  const value = await resolveInternal("page", scope).$eval(
+    selector,
+    jsProcessing
+  );
+  if (value === null && !optional) {
+    throw new BrowseError({
+      message: `Element given to @${type} call had no content of the correct type. If this is allowable, try using @${type}?`,
```

Comment:
"If this value is optional, try using .... This will set the value to null if it can't find a match"